dibridge/irc.py

```diff
@@ -156,17 +156,24 @@ async def _send_message(self, discord_id, discord_username, message, is_action=F
             await self._puppets[discord_id].send_message(message)
 
     async def _relay_mesage(self, irc_username, message):
-        # Don't echo back talk done by our puppets.
         for discord_id, puppet in self._puppets.items():
+            # Don't echo back talk done by our puppets.
             if puppet._nickname == irc_username:
                 return
 
-            # On IRC, it is common to do "name: ", but on Discord you don't do that ": " part.
-            if message.startswith(f"{puppet._nickname}: "):
-                message = f"<@{discord_id}> " + message[len(f"{puppet._nickname}: ") :]
-
             # If the username is said as its own word, replace it with a Discord highlight.
-            message = re.sub(r"(?<!\w)" + re.escape(puppet._nickname) + r"(?!\w)", f"<@{discord_id}>", message)
+            message = " ".join(
+                [
+                    re.sub(r"(?<!\w)" + re.escape(puppet._nickname) + r"(?!\w)", f"<@{discord_id}>", part)
+                    if "://" not in part
+                    else part
+                    for part in message.split(" ")
+                ]
+            )
+
+            # On IRC, it is common to do "name: ", but on Discord you don't do that ": " part.
+            if message.startswith(f"<@{discord_id}>: "):
```

Comment:
I did a quick test in python interpreter, and this replacement seem to be already handled by the previous instruction.
```
>>> import re
>>> nickname = "glx[d]"
>>> message = "glx[d]: test http://glx[d].com/"
>>> message = " ".join([re.sub(r"(?<!\w)" + re.escape(nickname) + r"(?!\w)", f"HL", part) if "://" not in part else part for part in message.split(" ")])
>>> print(message)
HL: test http://glx[d].com/
```

Reply:
Please read the comment and execute the code. It should show a difference ;)

Reply:
Oh yes, I carefully read the code after commenting, it's indeed not the same replacement. But that implies IRC user know the discord id of the puppet.

Reply:
No? The replacement to discord is done first, then it checks if there is a : after the highlight?

Just ran some tests, seems to be doing exactly what it should? I am a bit confused, sorry :p